src/borg/helpers.py

```diff
@@ -1602,11 +1602,29 @@ def finish(self):
 
 
 def sysinfo():
+    orig_env = os.environ
+    cleaned_env = prepare_subprocess_env(system=True)
+    linux_distribution = None
+    try:
+        # some platform calls internally run /bin/sh, so run them within a cleaned env,
+        # so that sh doesn't use pyinstaller env libs, but system libs, see #3732
+        os.environ = cleaned_env
```

Comment:
not sure whether this is good enough, so it actually ends up in internally created subprocesses.